common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerTokenProvider.java

```diff
@@ -0,0 +1,77 @@
+package com.microsoft.identity.common.internal.broker;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+import com.microsoft.identity.common.exception.ClientException;
+import com.microsoft.identity.common.exception.ErrorStrings;
+import com.microsoft.identity.common.internal.logging.Logger;
+import com.microsoft.identity.common.internal.providers.microsoft.MicrosoftAuthorizationRequest;
+
+
+
+public class BrokerTokenProvider {
+
+    private static final String TAG = BrokerTokenProvider.class.getSimpleName();
+
+    /**
+     * Acquire token interactively, will prompt user if possible.
+     * See {@link MicrosoftAuthServiceHandler#getIntentForInteractiveRequest(final Context context)} for details.
+     */
+    public void acquireTokenInteractively(final Activity activity, MicrosoftAuthorizationRequest request)
+            throws ClientException {
+        final String methodName = ":acquireTokenInteractively";
+
+        if (activity == null) {
+            throw new ClientException(ErrorStrings.EMPTY_ANDROID_CONTEXT);
+        }
+
+        final Bundle requestBundle = getBrokerOptions(request);
+        final Intent brokerIntent = MicrosoftAuthServiceHandler.getInstance().getIntentForInteractiveRequest(activity);
+        if (brokerIntent == null) {
+            ClientException exception = new ClientException(ErrorStrings.BROKER_APP_NOT_RESPONDING, "Received null intent from broker interactive request.");
+            Logger.error(TAG, "Received null intent from broker interactive request.", exception);
+            throw exception;
+        } else {
+            brokerIntent.putExtras(requestBundle);
+            brokerIntent.putExtra(AuthenticationConstants.Broker.BROKER_REQUEST,
+                    AuthenticationConstants.Broker.BROKER_REQUEST);
+        }
+
+        Logger.verbose(TAG + methodName, "Calling activity. " + "Pid:" + android.os.Process.myPid()
+                + " tid:" + android.os.Process.myTid() + "uid:" + android.os.Process.myUid());
+        activity.startActivityForResult(brokerIntent, AuthenticationConstants.UIRequest.BROWSER_FLOW);
+    }
+
+
+    private Bundle getBrokerOptions(final MicrosoftAuthorizationRequest request) {
+        Bundle brokerOptions = new Bundle();
+        // request needs to be parcelable to send across process
+        brokerOptions.putString(AuthenticationConstants.Broker.ACCOUNT_AUTHORITY, request.getAuthority().toString());
+        brokerOptions.putString(AuthenticationConstants.Broker.ACCOUNT_RESOURCE, request.getAuthorizationEndpoint());
```

Comment:
Should be scope, there is no resource in MSAL

Reply:
sorry just saw scope is below, just remove resource

Reply:
done